edpop-record-ontology.ttl

```diff
@@ -0,0 +1,140 @@
+# rudimentally converted into RDF, so that it will be possible to show this data in the VRE.
+edpoprec:originalData a rdf:Property ;
+                      skos:prefLabel "Original data"@en ;
+		      skos:description "All data from the original catalogue record" ;
+		      rdfs:domain edpoprec:Record .
+
+### PROPERTIES SPECIFIC FOR BIBLIOGRAPHICAL RECORDS
+
+# So far we only define string literals as ranges for these properties, but we could think
+# of also allowing other classes, like our own biographical records for authors.
+
+# The string literals are meant to be the best effort of EDPOP Explorer to give a 
+# somehow normalized, human-readable representation of the property, but they are not
+# thesaurized and certainly not guaranteed to be the same if they are conceptually the same
+# (e.g., Köln or Keulen for the city of Cologne).
+
+# My proposal would be to use these properties that can be extracted from the original record
+# in a deterministic way. That means: only string literals, and also links to other records
+# if they are defined in the original catalogues (such as links to CT Thesaurus).
+# Users will be able to enrich, correct and further normalize this data with annotations.
+# Annotations may also be used, perhaps, for automatic non-deterministic normalizations,
+# such as the place of publication as a link to Wikidata.
+
+edpoprec:title a rdf:property ;
+               skos:prefLabel "Title"@en ;
+               rdfs:domain edpoprec:BibliographicalRecord ;
+               rdfs:range xsd:string .
+
+edpoprec:alternativeTitle a rdf:property ;
+                            skos:prefLabel "Alternative title"@en ;
+                            rdfs:domain edpoprec:BibliographicalRecord ;
+                            rdfs:range xsd:string .
+
+# Or: contributor? And how to define the role of the contributor? Perhaps a separate
+# class is better. The catalogues generally describe the roles of the contributors quite
+# well (author, translator, illustrator, etc.)
+edpoprec:author a rdf:property ;
+                skos:prefLabel "Author"@en ;
+		rdfs:domain edpoprec:BibliographicalRecord ;
+		rdfs:range xsd:string .
+
+edpoprec:publisherOrPrinter a rdf:property ;
+                            skos:prefLabel "Publisher or printer"@en ;
+			    rdfs:domain edpoprec:BibliographicalRecord ;
+			    rdfs:range xsd:string .
+
+edpoprec:placeOfPublication a rdf:property ;
+                            skos:prefLabel "Place of publication"@en ;
+			    rdfs:domain edpoprec:BibliographicalRecord ;
+			    rdfs:range xsd:string .
+
+# Dating has xsd:string as its range, because the date is very often not a simple year
+# in the catalogues but can be anything, like [1650] for an estimation or a range like
+# 1800-1914.
+edpoprec:dating a rdf:property ;
+                skos:prefLabel "Dating"@en ;
+		rdfs:domain edpoprec:BibliographicalRecord ;
+		rdfs:range xsd:string .
```

Comment:
Agreed that we should have something queryable for the normalized version of the data.

Reply:
Querying dates will for sure be essential, but normalizing a date is not trivial, so that would have to be recorded separately.

A normalized date class should be able to express a bit more, I think, apart from the min/max date (in case of long-lasting publications, like multi-volume publications):
* The fact that the date is not printed on the publication itself, but instead either inferred or guessed (usually indicated as in [1750]
* In case of uncertainty: a range of possible years
* (To be complete, we would also need to support non-Christian eras, but probably we won't need that in EDPOP)

Reply:
Turns out there is already a data format that supports all of this: the Extended Date/Time Format from LoC: https://www.loc.gov/standards/datetime/ (URI: http://id.loc.gov/datatypes/edtf/EDTF). All information is in a single string, like "1984?" = possibly 1984 and or "2004-06/2006-08" for a range between June 2004 and August 2006.

What we need though then is a way to convert to this type, but also a way to extract possible beginnings and endings for these dates, to facilitate searching. I have seen implementations in C# and PHP, but nothing yet in Python.